### PR TITLE
README.md rework

### PR DESCRIPTION
**Pull request by tbkka**
-> Moves sales pitch to the front
-> Clarifies the options for including the library
-> Expands the Quick Start with Swift client code examples

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -105,7 +144,7 @@ directory.
 To install, just copy this one executable into a directory that is
```
Comment (thomasvl) [inline README.md]: Right above here, should we tweak this to suggest building release and linking the runtime static like we did on the other page?

File: README.md [hunk 2]
```diff
@@ -181,50 +232,45 @@ message BookInfo {
...
+of serialization and deserialization capabilities:
+```swift
+// Create a BookInfo object and populate it:
+var info = BookInfo()
+info.id = 1734
+info.title = "Really Interesting Book"
+info.author = "Jane Smith"
+
+// As above, but generating a read-only value:
+let info2 = BookInfo. with {
```
Comment (thomasvl) [inline README.md]: drop the space in there? `BookInfo.with {`

File: README.md [hunk 3]
```diff
@@ -181,50 +232,45 @@ message BookInfo {
...
+    $0.id = 1735
+    $0.title = "Even More Interesting"
+    $0.author = "Jane Q. Smith"
+  }
+
+// Serialize to binary protobuf format:
+let binaryData: Data = info.serializedData()
```
Comment (thomasvl) [inline README.md]: I believe most of these need `try`, up to you if you want to also include examples of error handling.

Comment (tbkka) [inline README.md]: Ouch!  Good catch!  Fixed.

File: README.md (hunk 2 above)
Comment (tbkka) [inline README.md]: Done.

File: README.md (hunk 1 above)
Comment (tbkka) [inline README.md]: Done.

**Pull request merged**